Yoast/wordpress-seo
# Default configuration for the Modal component.

## Summary

This PR can be summarized in the following changelog entry:

*  [not-userfacing] Adds default configuration for the Modal component

## Relevant technical choices:

The main goal of this PR is trying to pass a sane default configuration to the Modal component, as was the case in the original implementation where the config was passed via PHP, see https://github.com/Yoast/wordpress-seo/blob/4d4cd94cccde2d4c040b6c4367b803633aaadfa1/admin/class-yoast-modal.php#L74

When using the Modal component, the defaults can be extended / overridden.

There are no other pending items from the related issue, as the unused PHP classes have been removed already.

- adds a config object with default props
- merges passed props with default props
- restores the default "X" close button missed in https://github.com/Yoast/wordpress-seo/pull/10678
- removes the `a11yNotice` label: seems to me it was not used: please double check this point
- reduces the ESLint warnings

Note:
I'm not 100% sure the translatable strings can work correctly this way: I seem to recall there was a problem when passing translatable strings as default props but I couldn't find a clear reference. /Cc @atimmer 

## Test instructions
- build the assets
- test in the classic editor metabox and also in the Gutenberg editor metabox and sidebar
- verify the links `+ Add synonyms` and `+ Add related keyphrase` open the related modals
- verify the modals have the "X" close button in the top right corner
- verify there are no visual regressions

Fixes #11051
